server/etcdserver/server.go

```diff
@@ -904,10 +904,26 @@ func (s *EtcdServer) revokeExpiredLeases(leases []*lease.Lease) {
 	})
 }
 
+// isActive checks if the etcd instance is still actively processing the
+// heartbeat message (ticks). It returns false if no heartbeat has been
+// received within 3 * tickMs.
```

Comment:
Just out of curiosity is there a particular reason for `3`, or was it simply a reasonable duration?

Reply:
It should be a reasonable value derived from experience, e.g. Keepalive interval & Keepalive retry. There are also some route protocol use similar retry value.